src/common/style/icon_color_css.ts

```diff
@@ -1,5 +1,39 @@
 import { css } from "lit-element";
 
+export interface StateColor {
+  state?: string | number;
+  color: string;
+  below?: number;
+}
+
+export const computeCustomStateColor = (
+  state: string | number,
+  stateColors: StateColor[]
+): string | undefined => {
+  let color;
+  const sortableStateColors = stateColors.filter(
+    (stateColor) => stateColor.below
+  );
+
+  sortableStateColors.sort(
+    (a: StateColor, b: StateColor) => (a.below ?? 0) - (b.below ?? 0)
+  );
+
+  for (const stateColor of sortableStateColors) {
+    if (stateColor.below! < Number(state)) {
+      color = stateColor.color;
+    }
+  }
+
+  for (const stateColor of stateColors) {
+    if (stateColor.state === state || stateColor.state === Number(state)) {
+      color = stateColor.color;
+    }
+  }
+
+  return color;
```

Comment:
I opted to return color at the end so that you could have an exact numbers state match even when you have number ranges defined